### PR TITLE
Updates to several plugins to move to the org.opensearch.dataprepper package

### PR DESCRIPTION
**Pull request by dlvenable**
### Description

Updates several plugins to the `org.opensearch.dataprepper` package:

* csv
* date
* log-generator
* mutate
* aggregate
* drop
* grok
* http
* key-value.

These are plugins that currently have no know work in-flight, so this should not be disruptive to current PRs. 

### Issues Resolved

Partially: #344
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/drop-events-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/drop/DropEventProcessorConfigTest.java [hunk 1]
```diff
@@ -29,7 +28,7 @@ void getWhen() {
 
     @Test
     void getHandleFailedEventsOption() {
-        assertThat(config.getHandleFailedEventsOption(), is(HandleFailedEventsOption.SKIP));
+        assertThat(config.getHandleFailedEventsOption(), CoreMatchers.is(HandleFailedEventsOption.SKIP));
```
Comment (oeyh) [inline data-prepper-plugins/drop-events-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/drop/DropEventProcessorConfigTest.java:31]: Why the change here? There's a static import for `CoreMatchers.is` already.

**Review by oeyh — COMMENTED** [review]
Thanks for the updates, and all the cleanups. Just one small thing:

File: data-prepper-plugins/drop-events-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/drop/DropEventProcessorConfigTest.java (hunk 1 above)
Comment (dlvenable) [reply data-prepper-plugins/drop-events-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/drop/DropEventProcessorConfigTest.java]: I'm not quite sure. I used IntelliJ's "Optimize Imports" here. I fixed it in the latest commit.

**Review by oeyh — APPROVED** [review]

**Review by asifsmohammed — APPROVED** [review]
